kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -369,7 +407,7 @@ protected Optional<LogSegment> getLogSegmentForVersion(
             // We can't construct a snapshot because the directory contained no usable commit
             // files... but we can't return Optional.empty either, because it was not truly empty.
             throw new RuntimeException(
-                String.format("Empty directory: %s", logPath)
+                String.format("No delta files found in the directory: %s", logPath)
```

Comment:
What is the difference between the check here vs the check on the line #386? Is it just the `fileOpt` present vs not? If yes, we may want to document it saying that the fileOpt not empty means - listing has been tried.

Also should this exception also part of the DeltaErrors?

Reply:
https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java#L189
https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java#L158

386: Directory is empty or the directory is missing
410: No delta files found but the directory is not empty

Not sure if we want to treat all these different scenarios differently but if so we should do a separate PR. I think this specific exception doesn't need to be user-facing since having a table with a `_delta_log` directory but only having non-delta files in said directory seems like an invalid state.